### PR TITLE
feat(pre-commit): install a base set of pre-commit hooks to automatically maintain and sanity check repo changes

### PR DESCRIPTION
**Pull request by azend**
This change is mainly boilerplate to install `pre-commit` hooks into the repo. I've installed some basic hooks as well as a super simple linting tool. I'd like to add `pint` and some other Laravel specific tools but it looks like `pre-commit` currently supports basically every language _but_ PHP because I imagine the PHP tooling community is pretty insular. Some quick searching shows that `pint` is an option (it was always an option but some options take more effort than others) but, given we're still figuring out the development workflow and `pint` requires an execution environment with composer installed, I'm going to leave this out until we've better defined our other goals and scope.

### CODE REVIEW

File: README.md
Comment (IanEdington) [inline README.md:74]: I really hate leaving this kind of thing up to a developer to install and manage. I know pre-commit is great, but I wonder if there are any options that integrate directly with Composer. Something like Husky for node.

Another way I could see it working is on a composer install hook.

I worry about leaving more toil for the next person who comes along to work on this project.

Comment (azend) [reply README.md:74]: I hate the approach they take as well. At this moment in time, there's only a few project maintainers who likely have the hooks installed. Additional contributors will have code reviewed before merge. I've experimented in the past with using `pre-commit` hooks as a GitHub check but it's not perfect. I think for now something is better than nothing.

Comment (IanEdington) [reply README.md:74]: > I think for now something is better than nothing.

I would disagree. I've found that less is often more in software. Without having a CI check in place and linting the repo first this is going to add noise to almost every PR.

**Review by IanEdington — APPROVED** [review]

**Pull request merged**
